hack/test-cmd.sh

```diff
@@ -1617,6 +1623,27 @@ __EOF__
   # Clean up
   kubectl delete deployment nginx-deployment "${kube_flags[@]}"
 
+  ### Set image of a deployment 
+  # Pre-condition: no deployment exists
+  kube::test::get_object_assert deployment "{{range.items}}{{$id_field}}:{{end}}" ''
+  # Create a deployment
+  kubectl create -f hack/testdata/deployment-revision1.yaml "${kube_flags[@]}"
+  kube::test::get_object_assert deployment "{{range.items}}{{$id_field}}:{{end}}" 'nginx-deployment:'
+  kube::test::get_object_assert deployment "{{range.items}}{{$deployment_image_field}}:{{end}}" "${IMAGE_DEPLOYMENT_R1}:"
+  # Set the deployment's image 
+  kubectl set image deployment nginx-deployment nginx="${IMAGE_DEPLOYMENT_R2}" "${kube_flags[@]}"
+  kube::test::get_object_assert deployment "{{range.items}}{{$deployment_image_field}}:{{end}}" "${IMAGE_DEPLOYMENT_R2}:"
+  # Set non-existing container should fail 
+  ! kubectl set image deployment nginx-deployment redis=redis "${kube_flags[@]}"
+  # Set image of deployments without specifying name 
+  kubectl set image deployments --all nginx="${IMAGE_DEPLOYMENT_R1}" "${kube_flags[@]}"
+  kube::test::get_object_assert deployment "{{range.items}}{{$deployment_image_field}}:{{end}}" "${IMAGE_DEPLOYMENT_R1}:"
+  # Set image of a deployment specified by file
+  kubectl set image -f hack/testdata/deployment-revision1.yaml nginx="${IMAGE_DEPLOYMENT_R2}" "${kube_flags[@]}"
```

Comment:
I think david's point about being able to use set to mutate stuff without hitting the server is something we should do from day one.  Can you add it so that if you specify -f and -o yaml that we 

a) don't fetch from server (latest=false, or whatever) on resource builder
b) do the dry run equiv

I.e. I should only have to do `kubectl set image -f somefile.yaml A=B -o yaml` to mutate a not-yet created resource